### PR TITLE
Update code to check recent reference runs

### PR DESCRIPTION
**Pull request by weaverba137**
This PR is related to #40, but does not necessarily close it.

* Add script to easily check reference runs.
* Refactor code to use `desiutil.log` instead of `warnings.warn`.
* Fixes to the data model to reflect the actual code.

I will have additional commits to add to this, but I wanted to get it into the pipeline.

### CODE REVIEW

**Comment by weaverba137** [comment]
Here are some notes on problems I've found so far that might need to be turned into other tickets:

- pix and simpix files are missing from reference runs.
- desi and gfa files are missing from reference runs.
- No description for zcatalog roll-up file.
- versions.json missing from reference runs.

**Comment by weaverba137** [comment]
I am now satisfied that the code and data models support (as best as possible) the `DESI_SPECTRO_(DATA|REDUX|SIM)` sections relative to recent reference runs.  There is only the outstanding question of `DESI_TARGET` files (#44).

**Comment by weaverba137** [comment]
The target and fiberassign files (#44) are now also included.

File: bin/check_minitest.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+[[ -f ${SCRATCH}/DESI_SPECTRO_SIM_${run}.log ]] && /bin/rm -f ${SCRATCH}/DESI_SPECTRO_SIM_${run}.log
+[[ -f ${SCRATCH}/DESI_TARGET_${run}.log ]] && /bin/rm -f ${SCRATCH}/DESI_TARGET_${run}.log && touch ${SCRATCH}/DESI_TARGET_${run}.log
+#
+# Run the tests.
+#
+check_model DESI_SPECTRO_DATA ${DESI_SPECTRO_DATA} > ${SCRATCH}/DESI_SPECTRO_DATA_${run}.log 2>&1
+check_model DESI_SPECTRO_REDUX ${DESI_SPECTRO_REDUX} > ${SCRATCH}/DESI_SPECTRO_REDUX_${run}.log 2>&1
+check_model DESI_SPECTRO_SIM ${DESI_SPECTRO_SIM} > ${SCRATCH}/DESI_SPECTRO_SIM_${run}.log 2>&1
+check_model -F DESI_TARGET/mtl.rst /global/projecta/projectdirs/desi/datachallenge/reference_runs/18.2a/targets/mtl.fits >> ${SCRATCH}/DESI_TARGET_${run}.log 2>&1
+check_model -F DESI_TARGET/sky.rst /global/projecta/projectdirs/desi/datachallenge/reference_runs/18.2a/targets/sky.fits >> ${SCRATCH}/DESI_TARGET_${run}.log 2>&1
```
Comment (sbailey) [inline bin/check_minitest.sh]: minor (for now): should the hardcoded "18.2a" in these lines also be "${run}"?

Comment (weaverba137) [reply bin/check_minitest.sh]: Indeed, I'll fix that quickly, then merge.

**Comment by weaverba137** [comment]
I'm going to merge this one and create new issues and PRs for tracking 18.3-specific changes.

**Pull request merged**